### ISSUE TITLE
ENH: Add dtype_backend parameter to to_datetime

### ISSUE THREAD

**Issue by wirable23** [pandas-dev/pandas#53952]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

```
>>> a = pd.Series([4194304], dtype="int64[pyarrow]")
>>> pd.to_datetime(a).dtype
dtype('<M8[ns]')
>>>
```

In the above example, passing in a pyarrow backed series to to_datetime returns a numpy backed series, in to_numeric, there's a dtype_backend parameter that allows preserving of pyarrow dtypes, it would be nice to have the same option in to_datetime.

### Feature Description

Same way as in to_numeric, adding a dtype_backend parameter.

### Alternative Solutions

N/A

**Comment by phofl** [comment]
Yep this sounds good. Same as for to_numeric basically. Pinging @MarcoGorelli

**Comment by MarcoGorelli** [comment]
sounds good - interested in submitting a pull request @wirable23 ?

**Comment by wirable23** [comment]
@MarcoGorelli yes! The issue I was facing was that I didn't see arrow compute functions for some functionality of to_datetime, for example, pyarrow does not seem to have a compute function for creating datetimes from day month year components. Also, pyarrow does not support casting floating points to datetimes but there might be some other compute function that can be used.

**Comment by ruimamaral** [comment]
@lithomas1 @wirable23 I see there haven't been any updates to this issue in a while.
If you're both okay with it, I would like to implement this functionality.
Thanks!

**Comment by lithomas1** [comment]
Go for it.

**Comment by ruimamaral** [comment]
take

**Comment by ruimamaral** [comment]
I initially assigned myself as I thought I had a good idea of how this should be implemented, however after looking more deeply into to_datetime and pyarrow, I realize that I might've bit off more than I could chew...

My main issue is trying to figure out the best places to convert the dates between their numpy and pyarrow counterparts.
Since I don't feel certain about the full implementation details, I think it would be best to leave this one out to someone more familiar with pyarrow and this part of the codebase.

Therefore, I will be unassigning myself from this issue.

I apologize for any inconvenience caused!

**Comment by jbrockmendel** [comment]
I'm -1 on this, and actually would like to deprecate the keyword in to_numeric.  convert_dtypes should be The One Obvious Way to do this conversion.

**Comment by mattharrison** [comment]
Is there a canonical example of using `convert_dtypes` to pyarrow dates?

**Comment by berasaikat** [comment]
This worked for me - 

```python
s = pd.Series(["2023-01-01", "2023-06-15", "2024-03-20"])
result = pd.to_datetime(s).convert_dtypes(dtype_backend="pyarrow")
print(result.dtype)
``` 

output:
```
timestamp[us][pyarrow]
```
---

```python
df = pd.DataFrame({"date": ["2023-01-01", "2023-06-15"], "value": [1, 2]})
df["date"] = pd.to_datetime(df["date"])
df = df.convert_dtypes(dtype_backend="pyarrow")
print(df.dtypes)
```

output:
```
date     timestamp[us][pyarrow]
value            int64[pyarrow]
dtype: object
```
